### ISSUE TITLE
Any chance for SVG support?

### ISSUE THREAD

**Issue by mpbagot** [renpy/renpy#1641]
I've recently discovered renpy, and whilst looking through the documentation, I found that while it supports png and jpg formats for sprites, it doesn't have any support for vector formats (such as svg).

Are there any plans for support of vector graphics in the future?
If it's a desirable feature, I would be happy to attempt to implement it. Though I would appreciate any tips about what would be required for such a change.

**Comment by renpytom** [comment]
Basically, I'd need something that can either rasterize SVG, or something that could render it into OpenGL ES. And I'd need something that is portable, at least to windows, mac, linux, android, and iOS. Something that is relatively freely licensed (the LGPL being the most restrictive license Ren'Py has). 

And something that doesn't require a large set of libraries (like cairo, pango, glib, etc).

I don't know of any svg libraries that are like this, but I haven't looked in the last few years.

**Comment by renpytom** [comment]
Actually, it looks like SDL_image got some support for SVG, so it might be as easy as upgrading that dependency on all the platforms.

**Comment by mpbagot** [comment]
I've tested version 2.0.3 of SDL_image, and it does indeed give Renpy the ability to load svg files. The only downside is that the filename needs to be explicitly defined, since Renpy only looks for jpeg and png files.

Is there any chance for svg to be added as an autodetected file format for character or background images?

**Comment by Gouvernathor** [comment]
I'd like to bump on this, just to help reducing the tremendous file heaviness of some games...

**Comment by mcoirad** [comment]
@mpbagot How did you get renpy to successfully load SVG images? To my knowledge Renpy loads images using pygame.image which doesn't support SVG. 

[The referenced pygame docs](https://www.pygame.org/docs/ref/image.html#pygame.image.load)

**Comment by mpbagot** [comment]
I didn't actually look through any of RenPy's code. All I did was simply install libsdl2-image-dev v2.0.3 from the Ubuntu repos, and then build the RenPy modules as explained in the README. Then, to test, I simply placed an SVG into the images folder of the_question, and modified script.rpy.

To actually display the SVG image, I simply declared the image with:
```renpy
image test_svg = "test.svg"
```

And then to show it, added in a line:
```renpy
show test_svg
```
